### PR TITLE
Feature: adding `Huff` language support to playground code editor.

### PR DESCRIPTION
**Pull request by Saw-mon-and-Natalie**
This PR adds support for [Huff language](https://github.com/huff-language) (a low-level EVM language).

- [x] `yarn lint`
- [x] `yarn dev` functional

### CODE REVIEW

**Comment by dorlevi** [comment]
Thanks for the initiative! I know there are few people who would be happy with this addition, let us look at this in the next couple of days.

D

cc: @malexw @peroket

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "gray-matter": "^4.0.3",
     "highlight.js": "^11.3.1",
     "highlightjs-solidity": "^2.0.1",
+    "huffc": "Saw-mon-and-Natalie/huffc.git#fileless-parser",
```
Comment (peroket) [inline package.json:30]: If anything, why not point to the original repo?

**Review by peroket — COMMENTED** [review]
Thank you for the PR ;-)

I must admit I am discovering Huff with this PR, and I would love to know how you are planning to use this new feature with evm.codes.

I will also let @malexw to evaluate it technically.

File: package.json (hunk 1 above)
Comment (Saw-mon-and-Natalie) [reply package.json:30]: The original repo only compiles files and not the content as strings. It also misses declaration files for typescript. 

Once those has been added/modified in the original repo, we can change this to `huffc`.

Comment (Saw-mon-and-Natalie) [reply package.json:30]: I've created a PR for `huffc` to include the required changes:
[feat: Allow parsing/compiling of content only data ...](https://github.com/huff-language/huffc/pull/40)

Once that PR is reviewed and merged, we can update this `dependency`.

**Comment by Saw-mon-and-Natalie** [comment]
> Thank you for the PR ;-)
> 
> I must admit I am discovering Huff with this PR, and I would love to know how you are planning to use this new feature with evm.codes.
> 
> I will also let @malexw to evaluate it technically.

@peroket , Huff like other low-level languages (`YUL`) allows more direct access to EVM opcodes which can be used to optimize for gas consumption. It also includes some higher-level syntax like `macros` which allows writing `huff` programs more efficiently. 

Adding support for this language to evm.codes would be a great resource for teaching how to use the language and also to onboard new developers.

File: package.json (hunk 1 above)
Comment (peroket) [reply package.json:30]: That's great thanks :-)

**Comment by dorlevi** [comment]
@Saw-mon-and-Natalie : any news on getting https://github.com/huff-language/huffc/pull/40 in ? (we would like to merge this change but prefer to do so after this is in and the dependency is changed)

**Comment by Saw-mon-and-Natalie** [comment]
@dorlevi amazing. I'll ping the team Huff.

**Comment by dorlevi** [comment]
Sure thing, hope you understand why we want to wait for that to get resolved, and thanks again for contributing.

**Comment by dorlevi** [comment]
@Saw-mon-and-Natalie Any news on merging [the huf](https://github.com/huff-language/huffc/pull/40) change?

**Comment by Saw-mon-and-Natalie** [comment]
> @Saw-mon-and-Natalie Any news on merging [the huf](https://github.com/huff-language/huffc/pull/40) change?

Hi 👋🏽,

My PR was never reviewed on the `huffc` repo and the repo is now archived. In order to use the new implementation in Rust `huff-rs`, We/I would need to submit a new PR to the new repo to add a wasm target.

**Comment by peroket** [comment]
> > @Saw-mon-and-Natalie Any news on merging [the huf](https://github.com/huff-language/huffc/pull/40) change?
> 
> Hi @dorlevi 👋🏽,
> 
> My PR was never reviewed on the `huffc` repo and the repo is now archived. In order to use the new implementation in Rust `huff-rs`, We/I would need to submit a new PR to the new repo to add a wasm target.

Thank you for your anwser. Are you going to update your PR with the new one? We can't add a dependency if the repo is not going to be maintained unfortunately.

**Comment by ncitron** [comment]
I've actually been working on wasm bindings for the huff compiler. Once we've got them working I'd be happy to submit a PR if you don't mind @Saw-mon-and-Natalie?

**Comment by Saw-mon-and-Natalie** [comment]
> I've actually been working on wasm bindings for the huff compiler. Once we've got them working I'd be happy to submit a PR if you don't mind @Saw-mon-and-Natalie?

@ncitron LFG ☺️

**Comment by dorlevi** [comment]
@Saw-mon-and-Natalie and @ncitron, I'm just doing some housekeeping and wanted to check in on this.

**Pull request closed without merging**